qiskit/transpiler/passes/basis/decompose.py

```diff
@@ -58,20 +60,28 @@ def run(self, dag: DAGCircuit) -> DAGCircuit:
             output dag where ``gate`` was expanded.
         """
         # We might use HLS to synthesize objects that do not have a definition
-        hls = HighLevelSynthesis() if self.apply_synthesis else None
+        hls = HighLevelSynthesis(qubits_initially_zero=False) if self.apply_synthesis else None
```

Comment:
Was this a previous oversight?

Reply:
Yeah, we should not assume the initial qubit state since HLS here is called on individual blocks inside the circuit. However, technically, it didn't really matter since this argument only makes a difference if there are auxiliary qubits involved and the way we build the `node_as_dag` here won't have any auxiliary qubits. But still it's likely better to have the correct setting here 🙂